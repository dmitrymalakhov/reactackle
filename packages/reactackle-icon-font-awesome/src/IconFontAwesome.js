import React from 'react';
import PropTypes from 'prop-types';
import { registerDefaultComponentTheme, noop } from 'reactackle-core';
import { IconFontAwesomeStyled } from './styles/IconFontAwesomeStyled';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('icon-font-awesome', componentTheme);

const propTypes = {
  /** Set icon's name (required for font-awesome icons */
  name: PropTypes.string,
  /** Turns on/off icon's border */
  border: PropTypes.bool,
  /** Make icon rounded */
  rounded: PropTypes.bool,
  /** Set icon size */
  size: PropTypes.oneOf(['inherit', 'small', 'normal', 'large']),
  /** Scale icon */
  sizeMultiplier: PropTypes.number,
  /** Set icon's color scheme */
  colorScheme: PropTypes.oneOf(['light', 'dark']),
  /** Set exact icon's color */
  color: PropTypes.string,
  /** Set icon's background color */
  backgroundColor: PropTypes.string,
  /** Set icon's flipping plane */
  flip: PropTypes.oneOf(['none', 'horizontal', 'vertical']),
  /** Rotate icon by some degree */
  rotate: PropTypes.number,
  /** Make icon spinning infinitely & smooth */
  spin: PropTypes.bool,
  /** Make icon spinning infinitely & with steps */
  pulse: PropTypes.bool,
  /**
   * Specify function to call on Icon click
   */
  onClick: PropTypes.func,
  /**
   * Specify the tab order of an element
   */
  tabIndex: PropTypes.number,
};
const defaultProps = {
  name: 'font-awesome',
  border: false,
  rounded: false,
  size: 'normal',
  sizeMultiplier: 1,
  colorScheme: 'dark',
  color: '',
  backgroundColor: '',
  flip: 'none',
  rotate: 0,
  spin: false,
  pulse: false,
  onClick: noop,
  tabIndex: -1,
};

export default function IconFontAwesome(props) {
  const className = `fa fa-${props.name}`;

  const attributes = {
    className,
    onClick: props.onClick,
  };

  if (props.tabIndex !== -1) attributes.tabIndex = props.tabIndex;

  return (
    <IconFontAwesomeStyled
      {...attributes}
      border={props.border}
      rounded={props.rounded}
      sizeMultiplier={props.sizeMultiplier}
      sizeKey={props.size}
      colorScheme={props.colorScheme}
      userColor={props.color}
      bgColor={props.backgroundColor}
      flip={props.flip}
      rotate={props.rotate}
      spin={props.spin}
      pulse={props.pulse}
    />
  );
};

IconFontAwesome.propTypes = propTypes;
IconFontAwesome.defaultProps = defaultProps;
IconFontAwesome.displayName = 'IconFontAwesome';
