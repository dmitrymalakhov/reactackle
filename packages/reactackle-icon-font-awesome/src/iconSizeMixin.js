export const iconFASizeMixin = (outerSize, imgSize, outerWidth = outerSize) =>`
  width: ${outerWidth};
  height: ${outerSize};
  line-height: ${outerSize} !important;
  font-size: ${imgSize}px !important;
`;
