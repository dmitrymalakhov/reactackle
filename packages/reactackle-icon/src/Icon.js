import React from 'react';
import PropTypes from 'prop-types';
import { registerDefaultComponentTheme, noop } from 'reactackle-core';
import componentTheme from './styles/theme';
import { IconStyled } from './styles/IconStyled';
import { HolderStyled } from './styles/HolderStyled';

registerDefaultComponentTheme('icon', componentTheme);

const propTypes = {
  /** Set icon's source */
  src: PropTypes.string,
  /** Turns on/off icon's border */
  border: PropTypes.bool,
  /** Make icon rounded */
  rounded: PropTypes.bool,
  /** Set icon size */
  size: PropTypes.oneOf(['inherit', 'small', 'normal', 'large']),
  /** Scale icon */
  sizeMultiplier: PropTypes.number,
  /** Set icon's background color */
  backgroundColor: PropTypes.string,
  /** Set icon's flipping plane */
  flip: PropTypes.oneOf(['none', 'horizontal', 'vertical']),
  /** Rotate icon by some degree */
  rotate: PropTypes.number,
  /** Make icon spinning infinitely & smooth */
  spin: PropTypes.bool,
  /** Make icon spinning infinitely & with steps */
  pulse: PropTypes.bool,
  /**
   * Specify function to call on Icon click
   */
  onClick: PropTypes.func,
  /**
   * Specify the tab order of an element
   */
  tabIndex: PropTypes.number,
  /** Set icon's color scheme */
  colorScheme: PropTypes.oneOf(['light', 'dark']),
};

const defaultProps = {
  src: 'data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4KPCEtLSBHZW5lcmF0b3I6IEFkb2JlIElsbHVzdHJhdG9yIDIxLjAuMiwgU1ZHIEV4cG9ydCBQbHVnLUluIC4gU1ZHIFZlcnNpb246IDYuMDAgQnVpbGQgMCkgIC0tPgo8c3ZnIHZlcnNpb249IjEuMSIgaWQ9ItCh0LvQvtC5XzEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IgoJIHZpZXdCb3g9IjAgMCAyNCAyNCIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgMjQgMjQ7IiB4bWw6c3BhY2U9InByZXNlcnZlIj4KPHN0eWxlIHR5cGU9InRleHQvY3NzIj4KCS5zdDB7ZmlsbDpub25lO3N0cm9rZTojMDAwMDAwO3N0cm9rZS1saW5lY2FwOnJvdW5kO3N0cm9rZS1saW5lam9pbjpyb3VuZDtzdHJva2UtbWl0ZXJsaW1pdDoxMDt9Cjwvc3R5bGU+Cjxwb2x5Z29uIGNsYXNzPSJzdDAiIHBvaW50cz0iMjMuNSwwLjUgNS41LDAuNSA1LjUsMTguNSAyMy41LDE4LjUgIi8+CjxyZWN0IHg9IjAuNSIgeT0iNS41IiBjbGFzcz0ic3QwIiB3aWR0aD0iMTgiIGhlaWdodD0iMTgiLz4KPGc+Cgk8bGluZSBjbGFzcz0ic3QwIiB4MT0iMC41IiB5MT0iNS41IiB4Mj0iNS41IiB5Mj0iMC41Ii8+CjwvZz4KPGxpbmUgY2xhc3M9InN0MCIgeDE9IjIzLjUiIHkxPSIxOC41IiB4Mj0iMTguNSIgeTI9IjIzLjUiLz4KPGc+Cgk8bGluZSBjbGFzcz0ic3QwIiB4MT0iMTguNSIgeTE9IjUuNSIgeDI9IjIzLjUiIHkyPSIwLjUiLz4KPC9nPgo8L3N2Zz4K',
  border: false,
  rounded: false,
  size: 'normal',
  sizeMultiplier: 1,
  backgroundColor: '',
  flip: 'none',
  rotate: 0,
  spin: false,
  pulse: false,
  onClick: noop,
  tabIndex: -1,
  colorScheme: 'dark',
};

export default function Icon(props) {
  const attributes = {
    onClick: props.onClick,
  };

  if (props.tabIndex !== -1) attributes.tabIndex = props.tabIndex;

  return (
    <IconStyled
      {...attributes}
      border={props.border}
      rounded={props.rounded}
      sizeMultiplier={props.sizeMultiplier}
      sizeKey={props.size}
      bgColor={props.backgroundColor}
      colorScheme={props.colorScheme}
      flip={props.flip}
      rotate={props.rotate}
      spin={props.spin}
      pulse={props.pulse}
    >
      <HolderStyled
        sizeMultiplier={props.sizeMultiplier}
        sizeKey={props.size}
        src={props.src}
      />
    </IconStyled>
  );
};

Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;
Icon.displayName = 'Icon';
