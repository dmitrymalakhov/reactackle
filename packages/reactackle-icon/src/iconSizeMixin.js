export const iconSizeMixin = (outerSize, imgSize, outerWidth = outerSize) =>`
  width: ${outerWidth};
  height: ${outerSize};
  line-height: ${outerSize} !important;
  
  & > * {
    background-size: ${imgSize};
  }
`;
