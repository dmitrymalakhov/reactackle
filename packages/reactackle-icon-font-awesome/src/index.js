import IconFontAwesome from './IconFontAwesome';

export { IconFontAwesome };
export * from './iconSizeMixin';
