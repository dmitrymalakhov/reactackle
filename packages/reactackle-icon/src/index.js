import Icon from './Icon';

export { Icon };
export * from './iconSizeMixin';
