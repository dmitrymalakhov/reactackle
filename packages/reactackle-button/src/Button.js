'use strict';

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'reactackle-icon';
import { noop, registerDefaultComponentTheme } from 'reactackle-core';

import { ButtonStyled } from './styles/ButtonStyled';
import { ButtonTextStyled } from './styles/ButtonTextStyled';
import { ButtonTitleStyled } from './styles/ButtonTitleStyled';
import { ButtonSubtitleStyled } from './styles/ButtonSubtitleStyled';
import { ButtonIconBoxStyled } from './styles/ButtonIconStyled';

import componentTheme from './styles/theme';

registerDefaultComponentTheme('button', componentTheme);

const DefaultIcon = ({ size }) => (
  <Icon size={size === 'inline' ? 'inherit' : size} />
);

const propTypes = {
  /** Define button's main text */
  text: PropTypes.string,
  /**
   * Define button's alternate text for the user,
   * if he/she for some reason can't see the component
   */
  alternateTitle: PropTypes.string,
  /** Define button's subtitle */
  subtitle: PropTypes.string,
  /** Define button's size */
  size: PropTypes.oneOf(['inline', 'small', 'normal', 'large']),
  /** Narrow button has smaller horizontal paddings */
  narrow: PropTypes.bool,
  /** Define button's color */
  colorScheme: PropTypes.oneOf([
    'primary',
    'secondary',
    'alert',
    'success',
    'warning',
    'info',
    'flat',
    'flatLight',
    'white',
    'link',
  ]),
  /** Outlined button has border and transparent background */
  outlined: PropTypes.bool,
  /** Define button's radius */
  radius: PropTypes.oneOf(['none', 'default', 'rounded']),
  /** Raised button has shadow */
  raised: PropTypes.bool,
  /** Set disable to true if interaction with button isn't available to user */
  disabled: PropTypes.bool,
  /** Swap icon and text position */
  iconPositionRight: PropTypes.bool,
  /**
   * Turn on icon
   */
  hasIcon: PropTypes.bool,
  /**
   * Define icon component
   */
  iconComponent: PropTypes.element,
  /**
   * Stop onClick event's propagation and only call onPress
   */
  stopPressPropagation: PropTypes.bool,
  /**
   * Specify function to call on Button press
   */
  onPress: PropTypes.func,
  /**
   * The URL to link to when the button is clicked
   */
  href: PropTypes.string,
};

const defaultProps = {
  text: '',
  alternateTitle: '',
  subtitle: '',
  size: 'normal',
  narrow: false,
  colorScheme: 'flat',
  outlined: false,
  radius: 'default',
  raised: false,
  disabled: false,
  iconPositionRight: false,
  stopPressPropagation: false,
  hasIcon: false,
  iconComponent: DefaultIcon,
  onPress: noop,
  href: '',
};

export default class Button extends Component {
  constructor(props) {
    super(props);

    this._handleClick = this._handleClick.bind(this);
  }

  _handleClick(event) {
    if (!this.props.disabled) {
      if (this.props.stopPressPropagation) event.stopPropagation();
      this.props.onPress(event);
    }
  }

  render() {
    const subtitle =
      this.props.text && this.props.subtitle
        ? <ButtonSubtitleStyled size={this.props.size}>
            {this.props.subtitle}
          </ButtonSubtitleStyled>
        : null;

    const buttonText = this.props.text
      ? <ButtonTextStyled
          text={this.props.text}
          subtitle={this.props.subtitle}
          size={this.props.size}
          iconPositionRight={this.props.iconPositionRight}
        >
          {subtitle}
          <ButtonTitleStyled size={this.props.size}>
            {this.props.text}
          </ButtonTitleStyled>
        </ButtonTextStyled>
      : null;

    const IconComponent = this.props.iconComponent;
    const buttonIcon = this.props.hasIcon
        ? <ButtonIconBoxStyled
            iconPositionRight={this.props.iconPositionRight}
            size={this.props.size}
          >
            <IconComponent
              size={this.props.size}
            />
          </ButtonIconBoxStyled>
        : null;

    return (
      <ButtonStyled
        colorScheme={this.props.colorScheme}
        size={this.props.size}
        narrow={this.props.narrow}
        radius={this.props.radius}
        disabled={this.props.disabled}
        outlined={this.props.outlined}
        text={this.props.text}
        title={this.props.alternateTitle}
        subtitle={this.props.subtitle}
        raised={this.props.raised}
        onClick={this._handleClick}
        href={this.props.href}
      >
        {buttonIcon}
        {buttonText}
      </ButtonStyled>
    );
  }
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
Button.displayName = 'Button';
