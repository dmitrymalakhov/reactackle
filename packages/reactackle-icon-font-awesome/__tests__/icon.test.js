'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import { Icon } from '../src';

jest.mock('react-dom');

describe('<Icon/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <Icon name="github" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop name set', () => {
    const tree = renderer.create(
      <Icon name="github" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and size set inherit', () => {
    const tree = renderer.create(
      <Icon name="github" size="inherit" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and size set small', () => {
    const tree = renderer.create(
      <Icon name="github" size="small" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and size set normal', () => {
    const tree = renderer.create(
      <Icon name="github" size="normal" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and size set large', () => {
    const tree = renderer.create(
      <Icon name="github" size="large" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and spin set', () => {
    const tree = renderer.create(
      <Icon name="github" spin />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and pulse set', () => {
    const tree = renderer.create(
      <Icon name="github" pulse />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and border set', () => {
    const tree = renderer.create(
      <Icon name="github" border />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
  
  it('renders correctly if prop name and flip set horizontal', () => {
    const tree = renderer.create(
      <Icon name="github" flip="horizontal" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and flip set vertical', () => {
    const tree = renderer.create(
      <Icon name="github" flip="vertical" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and rotate set 180', () => {
    const tree = renderer.create(
      <Icon name="github" rotate={180} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and rotate set 0', () => {
    const tree = renderer.create(
      <Icon name="github" rotate={0} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and rotate set 90', () => {
    const tree = renderer.create(
      <Icon name="github" rotate={90} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and rotate set 270', () => {
    const tree = renderer.create(
      <Icon name="github" rotate={270} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and tabIndex set 1', () => {
    const tree = renderer.create(
      // eslint-disable-next-line jsx-a11y/tabindex-no-positive
      <Icon name="github" tabIndex={1} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop name and tabIndex set 100', () => {
    const tree = renderer.create(
      // eslint-disable-next-line jsx-a11y/tabindex-no-positive
      <Icon name="github" tabIndex={100} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('one call "onClick" event after "click"', () => {
    const mockFn = jest.fn(),
      tree = renderer.create(
        <Icon name="github" onClick={mockFn} />,
            ).toJSON();

    tree.props.onClick();

    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('renders with "library" as type', () => {
    const tree = renderer.create(
      <Icon type="library" src="foo://bar" name="github" />,
            ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  
  it('renders correctly if prop type set library and tabIndex set 1', () => {
    const tree = renderer.create(
      // eslint-disable-next-line jsx-a11y/tabindex-no-positive
      <Icon type="library" tabIndex={1} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
