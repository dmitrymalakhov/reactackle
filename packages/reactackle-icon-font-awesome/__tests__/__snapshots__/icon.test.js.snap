// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon/> renders correctly if prop name and border set 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq bSqQPn"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and flip set horizontal 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq ccpZVT"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and flip set vertical 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq fzeBvs"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and pulse set 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq goTdrN"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and rotate set 0 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq gTVhis"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and rotate set 90 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq gzSfsE"
  color=""
  onClick={[Function]}
  rotate={90}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and rotate set 180 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq jUHIyh"
  color=""
  onClick={[Function]}
  rotate={180}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and rotate set 270 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq exktAO"
  color=""
  onClick={[Function]}
  rotate={270}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and size set inherit 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq jIowIK"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and size set large 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq hHmaHZ"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and size set normal 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq gTVhis"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and size set small 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq ffLygF"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and spin set 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq cSzGrt"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop name and tabIndex set 1 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq gTVhis"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
  tabIndex={1}
/>
`;

exports[`<Icon/> renders correctly if prop name and tabIndex set 100 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq gTVhis"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
  tabIndex={100}
/>
`;

exports[`<Icon/> renders correctly if prop name set 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq gTVhis"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
/>
`;

exports[`<Icon/> renders correctly if prop type set library and tabIndex set 1 1`] = `null`;

exports[`<Icon/> renders correctly with default props 1`] = `
<span
  className="fa fa-github IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq gTVhis"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
/>
`;

exports[`<Icon/> renders with "library" as type 1`] = `
<div
  className="IconLibraryStyled__IconLibraryStyled-gcAxIp ejLgQW"
  color=""
  onClick={[Function]}
  rotate={0}
  size="normal"
  src=""
>
  <div
    className="IconHolderStyled__IconHolderStyled-bOzJUd jnpiax"
    color=""
    rotate={0}
    size="normal"
    src="foo://bar"
  />
</div>
`;
